src/createWidgetBase.ts

```diff
@@ -212,15 +196,35 @@ const createWidget: WidgetFactory = createStateful
 				return this.properties.id;
 			},
 
-			applyChangedProperties: function(this: Widget<WidgetState, WidgetProperties>, previousProperties: WidgetProperties, currentProperties: WidgetProperties): void {
-				if (Object.keys(currentProperties).length) {
-					currentProperties['id'] = this.id;
-					this.setState(currentProperties);
+			setProperties(this: Widget<WidgetState, WidgetProperties>, properties: WidgetProperties) {
```

Comment:
I think the rationale was:
* as a setter users may try to do `myWidget.properties.blah`, which wouldn't work
* as a setter although it does end up assigning to `this.properties` it runs a lifecycle before it; so isn't a pure setter
* the function matches `setState` in symmetry